crates/transaction-pool/src/lib.rs

```diff
@@ -12,6 +12,117 @@
+//! Transactions or NewPooledTransactionHashes message
+//!     ↓
+//! TransactionsManager (crates/net/network/src/transactions/mod.rs)
+//!     │
+//!     ├─→ For Transactions message:
+//!     │   ├─→ Validates message format
+//!     │   ├─→ Checks if transaction already known
+//!     │   ├─→ Marks peer as having seen the transaction
+//!     │   └─→ Queues for import
+//!     │
+//!     └─→ For NewPooledTransactionHashes message:
+//!         ├─→ Filters out already known transactions
+//!         ├─→ Queues unknown hashes for fetching
+//!         ├─→ Sends GetPooledTransactions request
+//!         ├─→ Receives PooledTransactions response
+//!         └─→ Queues fetched transactions for import
+//!             ↓
+//! pool.add_external_transactions() [Origin: External]
+//!     ↓
+//! Transaction Validation & Pool Addition
+//! ```
+//!
+//! ### 2. RPC Path (Local submission)
+//!
+//! ```text
+//! eth_sendRawTransaction RPC call
+//!     ├─→ Decodes raw bytes
+//!     └─→ Recovers sender
+//!         ↓
+//! pool.add_transaction() [Origin: Local]
+//!     ↓
+//! Transaction Validation & Pool Addition
+//! ```
+//!
+//! ### Transaction Origins
+//!
+//! - **Local**: Transactions submitted via RPC (trusted, may have different fee requirements)
+//! - **External**: Transactions from network peers (untrusted, subject to stricter validation)
+//! - **Private**: Local transactions that should not be propagated to the network
+//!
+//! ## Validation Process
+//!
+//! ### Stateless Checks  
+//!
+//! Ethereum transactions undergo several stateless checks:
+//!
+//! - **Transaction Type**: Fork-dependent support (Legacy always, EIP-2930/1559/4844/7702 need
+//!   activation)
+//! - **Size**: Input data ≤ 128KB (default)
+//! - **Gas**: Limit ≤ block gas limit
+//! - **Fees**: Priority fee ≤ max fee; local tx fee cap; external minimum priority fee
+//! - **Chain ID**: Must match current chain
+//! - **Intrinsic Gas**: Sufficient for data and access lists
+//! - **Blobs** (EIP-4844): Valid count, KZG proofs
+//!
+//! ### Stateful Checks
+//!
+//! 1. **Sender**: No bytecode (unless EIP-7702 delegated in Prague)
+//! 2. **Nonce**: ≥ account nonce
+//! 3. **Balance**: Covers value + (`gas_limit` × `max_fee_per_gas`)
+//!
+//! ### Common Errors
+//!
+//! - `NonceNotConsistent`: Nonce too low
+//! - `InsufficientFunds`: Insufficient balance
+//! - `ExceedsGasLimit`: Gas limit too high
+//! - `SignerAccountHasBytecode`: EOA has code
+//! - `Underpriced`: Fee too low
+//! - Blob errors: Missing sidecar, invalid proofs
+//!
+//! ## Subpool Design
+//!
+//! The pool maintains four distinct subpools, each serving a specific purpose
+//!
+//! ### Subpools
+//!
+//! 1. **Pending**: Ready for inclusion (no gaps, sufficient balance/fees)
+//! 2. **Queued**: Future transactions (nonce gaps or insufficient balance)
+//! 3. **`BaseFee`**: Valid but below current base fee
+//! 4. **Blob**: EIP-4844 transactions not pending due to insufficient base fee or blob fee
```

Comment:
this is inconsistent (`BaseFee` vs Pending/Blob, missing ticks)

Reply:
this is a clippy thing unfortunately

Reply:
ah oki